Python/ceval.c

```diff
@@ -230,6 +232,7 @@ PyEval_AcquireThread(PyThreadState *tstate)
     /* Check someone has called PyEval_InitThreads() to create the lock */
     assert(gil_created());
     take_gil(tstate);
+    exit_thread(tstate);
```

Comment:
`PyEval_RestoreThread` should also be switched over to using the new helper function.